### PR TITLE
Enable overload(operator.getitem)

### PR DESCRIPTION
**Pull request by sklam**
- [x] modify lowering to use `operator.getitem` as the key instead of `"getitem"`
- [x] modify typing to do the same as above
- [x] remove hack in lowering
- [x] add test

### CODE REVIEW

File: numba/typing/arraydecl.py [hunk 1]
```diff
@@ -159,9 +159,9 @@ def check_contiguity(outer_indices):
     return Indexing(idx, res, advanced)
 
 
-@infer
+@infer_global(operator.getitem)
 class GetItemBuffer(AbstractTemplate):
-    key = "getitem"
+    key = operator.getitem
```
Comment (seibert) [inline numba/typing/arraydecl.py]: Is it necessary to set the `key` attribute if you are using `infer_global`?

File: numba/typing/builtins.py [hunk 2]
```diff
@@ -550,9 +550,9 @@ def normalize_1d_index(index):
         return types.intp if index.signed else types.uintp
 
 
-@infer
+@infer_global(operator.getitem)
 class GetItemCPointer(AbstractTemplate):
-    key = "getitem"
+    key = operator.getitem
```
Comment (seibert) [inline numba/typing/builtins.py]: Same question here as above.

File: numba/typing/arraydecl.py (hunk 1 above)
Comment (sklam) [reply numba/typing/arraydecl.py]: I don't think so.  Let me try

**Comment by seibert** [comment]
ok, looks good.  Once this passes CI, it is ready to merge.

**Pull request merged**